source/triggers/clauses/clause_OnBlocklyPolygon.js

```diff
@@ -36,38 +36,50 @@ this.onGenerated = function( params, generator, payload ) {
     return true;
 }
 
-this.onPolygonFinished = function( blockNode, pointArray ) {
+this.onPolygonFinished = function( blockNode, playerPoints ) {
+
+    // Note: The playerPoints array should have length v+1 for the correct polygon. This is because we mark
+    // the point where the polygon is started and where it finishes to determine if it is closed.
 
     if ( this.blockNode === blockNode ) {
 
-        var equal = true;
-        var forward = true;
-        var backward = true;
+        // Does the player end up where they started? ( Closed Polygon )
 
-        for ( var i = 0; i < this.pointArray.length; i++ ) {
+        var startPoint = playerPoints[ 0 ];
+        var endPoint = playerPoints[ playerPoints.length - 1 ];
 
-                var thisPoint = this.pointArray[ i ];
-                var thatPoint = pointArray[ i ];
+        if ( startPoint[ 0 ] !== endPoint[ 0 ] || startPoint[ 1 ] !== endPoint[ 1 ] ) {
+            return;
+        }
 
-                if ( thisPoint[ 0 ] !== thatPoint[ 0 ] || thisPoint[ 1 ] !== thatPoint[ 1 ] ) { 
-                    forward = false;   
-                }           
+        // Are they the same length?
 
-                thisPoint = this.pointArray[ i ];
-                thatPoint = pointArray[ pointArray.length - 1 - i ];
+        if ( this.pointArray.length !== ( playerPoints.length - 1 )  ) {
+            return;
+        }
 
-                if ( thisPoint[ 0 ] !== thatPoint[ 0 ] || thisPoint[ 1 ] !== thatPoint[ 1 ] ) { 
-                    backward = false;   
-                }           
+        // Check if arrays are cyclic permutations of each other
+        // Concatenate the search array and try to find the playerPoints in that array
+        // Also check for the reverse
 
-        }
+        var forwardArray = this.pointArray;
+        var reversedArray = forwardArray.slice( 0 );
 
-        if ( this.pointArray.length !== pointArray.length ) {
-            equal = false;
-            return;
-        }
+        reversedArray.reverse();
+
+        var concatenatedForward = forwardArray.concat( forwardArray );
+        var concatenatedReverse = reversedArray.concat( reversedArray );
```

Comment:
Or, instead of doubling the array, just set your index counter to indexCounter % arraysize.  That will work as long as it doesn't go below zero.